Building without thread manager idle backoff fails

## Expected Behavior

Building with --DHPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=OFF used to just disable the idle manager backoff.  Now it breaks the build.

## Actual Behavior

```
In file included from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/runtime/resource/partitioner_fwd.hpp:11:0,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/runtime/resource/partitioner.hpp:10,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/runtime/resource/detail/partitioner.hpp:11,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/runtime/threads/executors/thread_pool_executors.hpp:12,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/include/threadmanager.hpp:11,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/include/runtime.hpp:15,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/apex/src/apex/apex.hpp:21,
                 from /var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/apex/src/apex/concurrency_handler.cpp:6:
/var/lib/buildbot/slaves/phylanx/x86_64-gcc7-debug/build/tools/buildbot/src/hpx/hpx/runtime/threads/policies/thread_queue_init_parameters.hpp:35:51: error: ‘HPX_IDLE_BACKOFF_TIME_MAX’ was not declared in this scope
             double max_idle_backoff_time = double(HPX_IDLE_BACKOFF_TIME_MAX),
                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~
```
See http://ktau.nic.uoregon.edu:8020/#/builders/7/builds/361/steps/5/logs/stdio for more info.

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Configure with -DHPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=OFF
  1. Make
  1. ...build fails

## Specifications

... Please describe your environment

  - HPX Version: current master/stable
  - Platform (compiler, OS): linux, gcc7

--- a/libs/config/include/hpx/config.hpp
+++ b/libs/config/include/hpx/config.hpp
@@ -421,11 +421,10 @@
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
-// Maximum sleep time for idle backoff in milliseconds.
-#if defined(HPX_HAVE_THREAD_MANAGER_IDLE_BACKOFF)
-#  if !defined(HPX_IDLE_BACKOFF_TIME_MAX)
-#    define HPX_IDLE_BACKOFF_TIME_MAX 1000
-#  endif
+// Maximum sleep time for idle backoff in milliseconds (used only if
+// HPX_HAVE_THREAD_MANAGER_IDLE_BACKOFF is defined).
+#if !defined(HPX_IDLE_BACKOFF_TIME_MAX)
+#  define HPX_IDLE_BACKOFF_TIME_MAX 1000
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////

